modules/hostap/src/supp_main.c

```diff
@@ -523,6 +585,455 @@ static int register_supplicant_event_socket(struct supplicant_context *ctx)
+	bss->ap_setup_locked = 1;
+#endif
+	conf->channel        = 1;
+	conf->acs            = conf->channel == 0;
+#ifdef CONFIG_ACS
+	conf->acs_num_scans  = 1;
+#endif /* CONFIG_ACS */
+	conf->ieee80211n     = 1;
+	conf->ieee80211h     = 0;
+	conf->ieee80211d     = 1;
+	conf->acs_exclude_dfs = 1;
+	conf->ht_capab |= HT_CAP_INFO_SHORT_GI20MHZ;
+	bss->auth_algs = 1;
+	bss->okc = 1;
+	conf->no_pri_sec_switch = 1;
+	conf->ht_op_mode_fixed  = 1;
+	conf->ieee80211ac       = 1;
+	conf->vht_oper_chwidth  = CHANWIDTH_USE_HT;
+	conf->ieee80211ax       = 1;
+	conf->he_oper_chwidth   = CHANWIDTH_USE_HT;
+	conf->he_phy_capab.he_su_beamformer = 0;
+	conf->he_phy_capab.he_su_beamformee = 1;
+	conf->he_phy_capab.he_mu_beamformer = 0;
+	conf->he_op.he_bss_color = 1;
+	conf->he_op.he_default_pe_duration  = 0;
+	/* Set default basic MCS/NSS set to single stream MCS 0-7 */
+	conf->he_op.he_basic_mcs_nss_set    = 0xfffc;
+
+	for (i = 0; i < conf->num_bss; i++) {
+		hostapd_set_security_params(conf->bss[i], 1);
+	}
+
+	if (hostapd_config_check(conf, 1)) {
+		errors++;
+	}
+
+#ifndef WPA_IGNORE_CONFIG_ERRORS
+	if (errors) {
+		wpa_printf(MSG_ERROR, "%d errors found in configuration file '%s'",
+			   errors, fname);
+		hostapd_config_free(conf);
+		conf = NULL;
+	}
+#endif /* WPA_IGNORE_CONFIG_ERRORS */
+	return conf;
+}
+
+static struct hostapd_iface *hostapd_interface_init(struct hapd_interfaces *interfaces,
+						    const char *if_name,
+						    const char *config_fname,
+						    int debug)
+{
+	struct hostapd_iface *iface;
+	int k;
+
+	wpa_printf(MSG_DEBUG, "Configuration file: %s", config_fname);
+	iface = hostapd_init(interfaces, config_fname);
+	if (!iface) {
+		return NULL;
+	}
+
+	if (if_name) {
+		os_strlcpy(iface->conf->bss[0]->iface, if_name,
+			   sizeof(iface->conf->bss[0]->iface));
+	}
+
+	iface->interfaces = interfaces;
+
+	for (k = 0; k < debug; k++) {
+		if (iface->bss[0]->conf->logger_stdout_level > 0) {
+			iface->bss[0]->conf->logger_stdout_level--;
+		}
+	}
+
+	if (iface->conf->bss[0]->iface[0] == '\0' &&
+	    !hostapd_drv_none(iface->bss[0])) {
+		wpa_printf(MSG_ERROR,
+			   "Interface name not specified in %s, nor by '-i' parameter",
+			   config_fname);
+		hostapd_interface_deinit_free(iface);
+		return NULL;
+	}
+
+	return iface;
+}
+
+static void zephyr_hostapd_init(struct supplicant_context *ctx)
+{
+	struct hapd_interfaces *interfaces = &ctx->hostapd;
+	size_t i;
+	int ret, debug = 0;
+	struct net_if *iface;
+	char ifname[IFNAMSIZ + 1] = { 0 };
+	const char *entropy_file = NULL;
+	size_t num_bss_configs = 0;
+	int start_ifaces_in_sync = 0;
+#ifdef CONFIG_DPP
+	struct dpp_global_config dpp_conf;
+#endif /* CONFIG_DPP */
+
+	os_memset(interfaces, 0, sizeof(struct hapd_interfaces));
+	interfaces->reload_config      = hostapd_reload_config;
+	interfaces->config_read_cb     = hostapd_config_read2;
+	interfaces->for_each_interface = hostapd_for_each_interface;
+	interfaces->ctrl_iface_init    = NULL;
+	interfaces->ctrl_iface_deinit  = NULL;
```

Comment:
This assignment is useless since you already called `memset()` (Probably you think it make sense to assign all the members of `interfaces`. However content of `interfaces` will probably change and the way the members are initialized will look illogical).

Reply:
Removed unnecessary assignment.